### DISCUSSION TITLE
Source Code Error 404

### DISCUSSION THREAD

**Discussion by ce4aser** [micropython/micropython#16009] [General]
I am confused. On page https://micropython.org/download/RPI_PICO2/ are uf2 files. But i want source code.

On this page is  Source on GitHub: rp2/RPI_PICO2 , but url is to https://github.com/micropython/micropython/tree/master/ports/rp2/boards/RPI_PICO2.

But error 404! Where is source code for RPi Pico 2 ??????

**Comment by robert-hh** [comment]
At the moment the source code for Pico 2 is available in the pull request #15619 for the RPi Pico port. The respective branch is
https://github.com/dpgeorge/micropython/tree/rp2-add-rp2350
You can as well check out the PR in a local copy of the MicroPython repository.

**Reply by ce4aser** [reply]
Thank you very much.

i have question.  On https://github.com/dpgeorge/micropython/blob/rp2-add-rp2350/ports/rp2/boards/RPI_PICO2/board.json is:

`   "variants": {
        "RISCV": "RISC-V CPU mode"
    },`

How to setup on ARM mode?

**Reply by robert-hh** [reply]
ARM mode is the default. RISCV mode is the variant, to be compiled with the option BOARD_VARIANT=RISCV.

**Reply by ce4aser** [reply]
> ARM mode is the default. RISCV mode is the variant, to be compiled with the option BOARD_VARIANT=RISCV.

Thank you very much.
